Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -511,14 +512,27 @@ static void paint_text_fragment(PaintContext& context, Layout::TextNode const& t
         DevicePixelPoint baseline_start { fragment_absolute_device_rect.x(), fragment_absolute_device_rect.y() + context.rounded_device_pixels(fragment.baseline()) };
         Utf8View view { text.substring_view(fragment.start(), fragment.length()) };
 
-        painter.draw_text_run(baseline_start.to_type<int>(), view, fragment.layout_node().font(), text_node.computed_values().color());
+        auto& font = fragment.layout_node().font();
+        auto scaled_font = [&]() {
+            auto device_font_pt_size = context.enclosing_device_pixels(font.presentation_size());
+            FontSelector font_selector = { font.family(), static_cast<float>(device_font_pt_size.value()), font.weight(), font.width(), font.slope() };
+            if (auto cached_font = FontCache::the().get(font_selector)) {
+                return cached_font;
+            }
+
+            auto font_with_device_pt_size = font.with_size(static_cast<float>(device_font_pt_size.value()));
+            FontCache::the().set(font_selector, *font_with_device_pt_size);
```

Comment:
`font_with_device_pt_size` is not guaranteed to be non-null, is it?

Reply:
right, thank you. `font_with_device_pt_size` is guaranteed to be non-null only for vector fonts but it might be null for bitmap fonts. I added handling for that case.